### ISSUE TITLE
[Feature]: Reflect the result of a test.step with the corresponding step in azure

### ISSUE THREAD

**Issue by joerg1985** [alexneo2003/playwright-azure-reporter#14]
As mentioned in #4 there are steps inside the azure testcases. It would be greate to connect the playwright `test.step` with the azure steps. To link these the same syntax like linking the testcases could be used, e.g.  `test.step("[2] login to the system")` -> is linked to the second step.

This is a minimal example of a playwright test using steps:
```
test.describe('google div', () => {
  test('[31] failing is enabled', async ({ page }) => {
    await test.step("[1] Just a step", async () => {
      await expect(true).toBe(true);
    });
    
    await test.step("[2] Just another step", async () => {
      await expect(page.locator("div")).toBeEnabled();
    });
  })
})
```

The result of the steps are reflected correctly to the HTML report, so it should be possible to get the status of a step.
This might be done using the [TestStep.error](https://playwright.dev/docs/api/class-teststep#test-step-error) field, this should be investigated.

**Comment by joerg1985** [comment]
The behaviour of https://playwright.dev/docs/test-assertions#soft-assertions should also be invstigated.

**Comment by alexneo2003** [comment]
at now i'm waiting for answer on my question which was asked at microsoft/azure-devops-node-api#545

**Comment by amitasil** [comment]
Eagerly waiting for this update!

**Comment by alexneo2003** [comment]
> Eagerly waiting for this update!

I'm sorry, but nothing depends on me now.  I am waiting for a response from the developers microsoft/azure-devops-node-api#545 about the possibility of publishing similar results.  and most likely such a possibility does not exist.

**Comment by joerg1985** [comment]
@alexneo2003 i had a look at the API docs and the `actionResults` are used to reflect test steps.
If you create a manual run with step results and look at it with the flag detailsToInclude=iterations set you can see the data.
.../_apis/test/Runs/{run.id}/Results/{result.id}?detailsToInclude=iterations

Based on the API docs i thinks you could set the `actionResults` to an array of [`TestActionResultModel`](https://learn.microsoft.com/en-us/rest/api/azure/devops/test/results/get?view=azure-devops-rest-7.0&tabs=HTTP#testactionresultmodel) to push the results of test steps.

**Comment by alexneo2003** [comment]
it's a good thing but when i'm add test results to test run i using [this](https://github.com/microsoft/azure-devops-node-api/blob/master/api/TestApi.ts#L58) method. as you seen it has `results: TestInterfaces.TestCaseResult[]` [argument](https://github.com/microsoft/azure-devops-node-api/blob/master/api/interfaces/TestInterfaces.ts#L2581) wich doesn't contain any implementation of `actionResults` about you say.
i'm see that ADO API [has](https://learn.microsoft.com/en-us/rest/api/azure/devops/test/results/add?view=azure-devops-rest-7.0&tabs=HTTP#testiterationdetailsmodel) this `actionResults` as parameter but azure-devops-node-api doesn't has any functionality to work with it, or i doesn't seen it.

**Comment by alexneo2003** [comment]
So, folks, we've try to implement this feature.
Unfortunaly I can't test it in "prod" :-( 
If you still interested in it - please test
The doc

### Test Step Results

Publish the outcome of each `test.step()` call as a step-level result inside the Azure DevOps test case result.

```typescript
{
  publishTestStepResults: true,
}
```

Steps are mapped to Azure test case steps by a numeric ID extracted from the step title using the `[N]` syntax (same convention as test case IDs). Steps without an explicit `[N]` are auto-numbered sequentially.

```typescript
test('[31] login flow', async ({ page }) => {
  await test.step('[1] navigate to login page', async () => {
    await page.goto('/login');
  });

  await test.step('[2] fill credentials', async () => {
    await page.fill('#username', 'user');
    await page.fill('#password', 'pass');
  });

  // No ID — auto-numbered as step 3
  await test.step('submit form', async () => {
    await page.click('#submit');
  });
});
```

Each step result includes: outcome (`Passed`/`Failed`), error message (if failed), and duration.

> **Note:** The numeric ID in `[N]` must match the step number in the Azure test case work item (steps are 1-based). The `actionPath` sent to Azure is computed as `(N + 1)` in 8-digit hex, which matches Azure's internal step ID format.


Package for instalation
```bash
npm install @alex_neo/playwright-azure-reporter@1.20.1-pr147.feat-publish-test-step-results.9ff5b71
```

Let me know about the results.
Thanks!

**Comment by alexneo2003** [comment]
So, I've tested it. In my situation it doesn't work, steps doesn't published into tests results and currently I don't know how to handle it 🤕 I leave issues opened

**Comment by smcthompson** [comment]
Howdy @alexneo2003 !  I have been working in the background on this feature.  I have working test step outcomes being published to an internal company ADO site:

<img width="396" height="892" alt="Image" src="https://github.com/user-attachments/assets/7b531611-7a8e-4a54-8f97-014f159929a7" />

<img width="459" height="294" alt="Image" src="https://github.com/user-attachments/assets/4d644bd6-93b0-4ec2-824e-1f909e0b6bd3" />

But I have the work-in-progress code [forked to my Github](https://github.com/smcthompson/playwright-azure-reporter/tree/feature/test-steps-outcome).  I had plans to include [Test Suite ID annotations](https://github.com/smcthompson/playwright-azure-reporter/blob/feature/test-steps-outcome/IMPLEMENTATION-SUITE-ID-ANNOTATIONS.md) for targeting test outcome reporting.  My work had stalled on implementing this as the company had moved in another direction with regards to reporting, but I am finally ready to pick that effort back up again!

**Comment by alexneo2003** [comment]
@smcthompson  I'll take a look at your implementation and if you want i can implement it here, if you don't mind

**Comment by smcthompson** [comment]
> [@smcthompson](https://github.com/smcthompson) I'll take a look at your implementation and if you want i can implement it here, if you don't mind

Not at all!!  It was my intention to eventually PR this work into your repo.  After reading this feature thread, I wanted you to kick the tires on what I had implemented to see if it aligned with your original goals.  My hope is that it fills the gap in spite of other challenges, and it turns out to be an "A-HA!!" moment for everyone who has been looking for this feature -- including yourself!

**Comment by alexneo2003** [comment]
> Not at all!! It was my intention to eventually PR this work into your repo. After reading this feature thread, I wanted you to kick the tires on what I had implemented to see if it aligned with your original goals. My hope is that it fills the gap in spite of other challenges, and it turns out to be an "A-HA!!" moment for everyone who has been looking for this feature -- including yourself!

to be honest, for the last almost 10 months I have been using python to write tests, it was not my choice 😢  and I have limited access to the typescript repository and even less access to the ability to test the reporter 🤕  so if you've done it and it works, and from what i've seen in the screenshots it really does, then this is exactly what was originally requested as a feature! therefore, I will only be glad, and I think not only me, if you contribute to the development of the project 👍
